JYRadarChart.m

```diff
@@ -93,39 +100,46 @@ - (void)setColors:(NSArray *)colors {
     }
 }
 
+- (void)setPointsColors:(NSMutableArray *)pointsColors{
+    _pointsColors = [NSMutableArray array];
+    for (UIColor *color in pointsColors) {
+        [_pointsColors addObject:[color colorWithAlphaComponent:self.pointsColorOpacity]];
+    }
+}
+
 - (void)setNeedsDisplay {
-	[super setNeedsDisplay];
-	[self.legendView sizeToFit];
-	[self.legendView setNeedsDisplay];
+    [super setNeedsDisplay];
+    [self.legendView sizeToFit];
+    [self.legendView setNeedsDisplay];
 }
 
-- (void)setDataSeries:(NSArray *)dataSeries {
-	_dataSeries = dataSeries;
-	NSArray *arr = _dataSeries[0];
-	_numOfV = [arr count];
-	if (self.legendView.colors.count < _dataSeries.count) {
-		for (int i = 0; i < _dataSeries.count; i++) {
-			UIColor *color = [UIColor colorWithHue:1.0 * (i * COLOR_HUE_STEP % MAX_NUM_OF_COLOR) / MAX_NUM_OF_COLOR
-			                            saturation:1
-			                            brightness:1
-			                                 alpha:self.colorOpacity];
-			self.legendView.colors[i] = color;
-		}
-	}
+- (void)setDataSeries:(NSMutableArray *)dataSeries {
```

Comment:
I think there's no point to make this the arrat mutable here while the property in the header file is still immutable